packages/template-retail-react-app/app/pages/account/wishlist/partials/wishlist-primary-action.jsx

```diff
@@ -55,44 +68,89 @@ const WishlistPrimaryAction = () => {
-                            onOpen={onOpen}
-                            onClose={onClose}
-                            product={variant}
-                            addToCart={(variant, quantity) => handleAddToCart(variant, quantity)}
-                        />
-                    )}
-                </>
-            ) : (
+    const buttonText = {
+        viewOptions: (
+            <FormattedMessage
+                defaultMessage="View Options"
+                id="wishlist_primary_action.button.view_options"
+            />
+        ),
+        addToCart: (
+            <FormattedMessage
+                defaultMessage="Add to Cart"
+                id="wishlist_primary_action.button.add_to_cart"
+            />
+        ),
+        addAllToCart: (
+            <FormattedMessage
+                defaultMessage="Add All to Cart"
+                id="wishlist_primary_action.button.add_all_to_cart"
+            />
+        )
+    }
+
+    let button
+
+    if (isProductASet) {
+        button = (
+            <Button
+                as={Link}
+                href={`/product/${variant.id}`}
+                w={'full'}
+                variant={'solid'}
+                _hover={{textDecoration: 'none'}}
+            >
+                {buttonText.viewOptions}
+            </Button>
+        )
+        // TODO: create a new set in BM to test this scenario
+        if (variant.setProducts?.every((child) => !hasVariants(child))) {
+            button = (
                 <Button
                     variant={'solid'}
                     onClick={() => handleAddToCart(variant, variant.quantity)}
                     w={'full'}
                     isLoading={isLoading}
                 >
-                    <FormattedMessage
-                        defaultMessage="Add to Cart"
-                        id="wishlist_primary_action.button.add_to_cart"
-                    />
+                    {buttonText.addAllToCart}
+                </Button>
+            )
+        }
+    } else {
+        button = isMasterProduct ? (
+            <>
+                <Button w={'full'} variant={'solid'} onClick={onOpen}>
+                    {buttonText.viewOptions}
                 </Button>
-            )}
-        </>
-    )
+                {isOpen && (
+                    <ProductViewModal
+                        isOpen={isOpen}
+                        onOpen={onOpen}
+                        onClose={onClose}
+                        product={variant}
+                        addToCart={(variant, quantity) => handleAddToCart(variant, quantity)}
+                    />
+                )}
+            </>
+        ) : (
+            <Button
+                variant={'solid'}
+                onClick={() => handleAddToCart(variant, variant.quantity)}
+                w={'full'}
+                isLoading={isLoading}
+            >
+                {buttonText.addToCart}
+            </Button>
+        )
+    }
+
+    return button
```

Comment:
Personally I like to avoid using `let` if at all possible. If you're open to it, I would consider just moving these to immediate returns like this

```javascript
if (isProductASet) {
  return <>{something}</>
} else if (condition) {
  return <>{somethingElse}</>
} else {
  return <>{default}</>
}
```

instead of doing `let button` and then declaring it in if / else block scope